hclsyntax/expression.go

```diff
@@ -845,6 +858,7 @@ func (e *ObjectConsExpr) Value(ctx *hcl.EvalContext) (cty.Value, hcl.Diagnostics
 			continue
 		}
 
+		key, _ = key.Unmark()
```

Comment:
I don't think this line is necessary, since you've already left the for loop when a marked key is found (the tests still pass without it).

Reply:
Oh thanks! Yep, that's a relic from when I thought about unmarking it and passing through, before I decided it should be an error.